### ISSUE TITLE
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Azure.Amqp, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.

### ISSUE THREAD

**Issue by deyanp** [Azure/azure-sdk-for-net#8479]
## Actual Behavior
1. When referencing Azure.EventHubs.dll in F# Interactive with the following:

```
#I "../../../../.nuget/packages/"
#r @"microsoft.azure.eventhubs\4.1.0\lib\netstandard2.0\Microsoft.Azure.EventHubs.dll"

```
and executing the following script

```
open Microsoft.Azure.EventHubs
open System.Text
open System.Threading.Tasks


let connStr = "Endpoint=sb://xxx-xxxx-xxxxx-hubs.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=xxxxxxxxxxxxxxxxxxxxxxxxxxxx="
let connStrBuilder = EventHubsConnectionStringBuilder(connStr)
connStrBuilder.EntityPath <- "testeventhub1"
let eventHubClient = EventHubClient.Create(connStrBuilder)
let message = "Test Message"
```

I am getting 

```
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Azure.Amqp, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
File name: 'Microsoft.Azure.Amqp, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at Microsoft.Azure.EventHubs.Amqp.AmqpEventHubClient..ctor(EventHubsConnectionStringBuilder csb, ITokenProvider tokenProvider)
```

## Expected Behavior
1. Reference and sample should work without such issues, or Microsoft.Azure.Amqp Version 2.4.3 or 2.4.2 should be referenced

## Versions  
- OS platform and version: Windows 10
- .NET Version: .Net Core 2.2, 3.0
- NuGet package version or commit ID: Nuget Microsoft.Azure.EventHubs 4.1.0

**Comment by serkantkaraca** [comment]
Can you also reference correct version of Microsoft.Azure.Amqp?

**Comment by deyanp** [comment]
How? There is no 2.4.0.0 on NuGet, and if I try with 4.2.3 or 4.2.2 it gives me the same error ..

**Comment by serkantkaraca** [comment]
4.1.0 AMQP library dependency is as below. So 2.4.2 or above should work.

Microsoft.Azure.Amqp (>= 2.4.2 && < 3.0.0)

**Comment by deyanp** [comment]
When I reference

> #I "../../../../.nuget/packages/"
> #r @"microsoft.azure.amqp\2.4.3\lib\netstandard1.3\Microsoft.Azure.Amqp.dll"
> #r @"microsoft.azure.eventhubs\4.1.0\lib\netstandard2.0\Microsoft.Azure.EventHubs.dll"
> 

i get the error

> File name: 'Microsoft.Azure.Amqp, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
>    at Microsoft.Azure.EventHubs.Amqp.AmqpEventHubClient..ctor(EventHubsConnectionStringBuilder csb, ITokenProvider tokenProvider)
>    at Microsoft.Azure.EventHubs.EventHubClient.Create(EventHubsConnectionStringBuilder csb)
>    at <StartupCode$FSI_0003>.$FSI_0003.main@()
> 
> 
> Stopped due to error

Did you try it yourself - you should be getting the same error ...

**Comment by deyanp** [comment]
This is what I get with ILSpy for Microsoft.Azure.EventHubs.dll 4.1.0, I think there is something wrong with the references:

// Detected Target-Framework-Id: .NETStandard,Version=v2.0

// Referenced assemblies (in metadata order):
// Microsoft.Azure.Amqp, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (unresolved)
// Microsoft.Azure.Services.AppAuthentication, Version=1.0.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (unresolved)
// Microsoft.IdentityModel.Tokens, Version=5.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (unresolved)
// netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
	// Assembly reference loading information:
	// Info: Success - Found in Assembly List

// System.Diagnostics.DiagnosticSource, Version=4.0.3.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
	// Assembly reference loading information:
	// Info: Success - Found in Assembly List

// System.IdentityModel.Tokens.Jwt, Version=5.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (unresolved)


// Assembly load log including transitive references:
// netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
	// Info: Success - Found in Assembly List

// System.Diagnostics.DiagnosticSource, Version=4.0.3.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
	// Info: Success - Found in Assembly List

**Comment by serkantkaraca** [comment]
Can you try building packages after moving the AMQP reference to up to date release? If works, I will work on releasing new SDK.

**Comment by deyanp** [comment]
Can you please elaborate your last request?
FYI, the above is a simple F# script, which is saved as a xyz.fsx file and executed with **dotnet fsi xyz.fsx** ... There is no package building or similar, you need to simply get restore the nuget packages in the referenced folder beforehand ...

Or do you mean I should create a compiled F# program, and see if that works?

**Comment by deyanp** [comment]
Tried with a compiled F# program, and it is OK. So the issue is only with references in fsx files..

**Comment by serkantkaraca** [comment]
This should have already fixed with 4.2.0 release.
